File: npm-package/bin/automaton-cli.js
#!/usr/bin/env node
/* automaton-cli — AI services CLI */
/* Usage: npx automaton-api-client review "function foo() { return 1; }" */

const API = 'https://automation.songheng.vip';

const HELP = `
Usage: npx automaton-api-client <service> [code|text]

Services:
  review     AI code review — bug detection, style issues, improvements
  security   Security vulnerability scan (OWASP Top 10)  
  analyze    Deep text analysis — sentiment, entities, themes
  summarize  Key points and takeaways
  explain    Plain language code explanation
  refactor   Code improvement suggestions
  complexity Time & space complexity analysis

Examples:
  npx automaton-api-client review "function add(a,b){return a+b}"
  npx automaton-api-client analyze "This product is amazing!"
  npx automaton-api-client security "eval(request.body.input)"
  
Options:
  --key KEY  Use premium API key (default: free tier)
  --json     Output raw JSON
  --help     Show this help

Free tier: 3 requests/day per service. 
Premium: https://automation.songheng.vip/upgrade
`;

const services = ['review', 'security', 'analyze', 'summarize', 'explain', 'refactor', 'complexity'];
const args = process.argv.slice(2);

if (args.includes('--help') || args.length === 0) {
  console.log(HELP);
  process.exit(0);
}

const service = args.find(a => !a.startsWith('--'));
const keyIndex = args.indexOf('--key');
const apiKey = keyIndex >= 0 ? args[keyIndex + 1] : null;
const jsonMode = args.includes('--json');

if (!service || !services.includes(service)) {
  console.error(`❌ Unknown service: "${service}"`);
  console.log(`Available: ${services.join(', ')}`);
  process.exit(1);
}

// Get the input (everything after the service name)
const serviceIndex = args.indexOf(service);
const inputParts = args.slice(serviceIndex + 1).filter(a => !a.startsWith('--') && (!apiKey || a !== args[keyIndex + 1]));
const input = inputParts.join(' ');

if (!input) {
  console.error('❌ Missing input text or code');
  process.exit(1);
}

const field = ['analyze', 'summarize'].includes(service) ? 'text' : 'code';
const body = { [field]: input };
const headers = { 'Content-Type': 'application/json' };
const url = apiKey
  ? `${API}/v1/${service}`
  : `${API}/api/free/${service}`;

if (apiKey) headers['X-API-Key'] = apiKey;

async function main() {
  try {
    const resp = await fetch(url, {
      method: 'POST',
      headers,
      body: JSON.stringify(body),
      signal: AbortSignal.timeout(30000)
    });

    const data = await resp.json();

    if (resp.status === 402) {
      console.log('❌ Insufficient credits. Get API key: ' + API + '/upgrade');
      process.exit(1);
    }
    if (resp.status === 429) {
      console.log('❌ Free limit reached (3/day). Upgrade: ' + API + '/upgrade');
      process.exit(1);
    }

    if (jsonMode) {
      console.log(JSON.stringify(data, null, 2));
    } else if (data.success) {
      console.log('\n' + data.result + '\n');
      if (data.credits_remaining !== undefined) {
        console.log(`Credits remaining: ${data.credits_remaining}`);
      }
    } else {
      console.error('❌ Error:', data.error);
      process.exit(1);
    }
  } catch (e) {
    console.error('❌ Connection error:', e.message);
    process.exit(1);
  }
}

main();

File: content/automaton-cli.py
#!/usr/bin/env python3
"""
my-automaton CLI — x402 AI API client
Usage:
  python3 automaton-cli.py analyze "your text here"
  python3 automaton-cli.py review --file /path/to/code.py
  python3 automaton-cli.py summarize --url https://example.com/article
  python3 automaton-cli.py --help

First 3 calls per day are FREE (no wallet needed).
After that, send USDC to 0x76eADdEBFfb6a61DD071f97F4508467fc55dd113 on Base chain.
"""
import sys
import json
import os
import urllib.request
import urllib.error

BASE_URL = "https://automation.songheng.vip"
WALLET = "0x76eADdEBFfb6a61DD071f97F4508467fc55dd113"
CHAIN = "Base"
TOKEN = "USDC"

SERVICES = {
    "analyze": {"cost": "1¢", "desc": "Deep text analysis — insights, patterns, key takeaways"},
    "summarize": {"cost": "2¢", "desc": "Concise AI summarization"},
    "review": {"cost": "5¢", "desc": "Full code review — bugs, style, security"},
    "security": {"cost": "3¢", "desc": "Security vulnerability scan (OWASP Top 10)"},
    "explain": {"cost": "2¢", "desc": "Code explanation — what it does and how"},
    "refactor": {"cost": "5¢", "desc": "Refactoring suggestions"},
    "complexity": {"cost": "2¢", "desc": "Big O complexity analysis"},
}

def call_api(service, text, code=None):
    """Call a service, trying free first, then prompting for payment."""
    data = {"text": text} if not code else {"code": code}
    
    # First try free
    req = urllib.request.Request(
        f"{BASE_URL}/free/{service}",
        data=json.dumps(data).encode(),
        headers={"Content-Type": "application/json"},
        method="POST"
    )
    
    try:
        with urllib.request.urlopen(req, timeout=30) as resp:
            result = json.loads(resp.read())
            if "result" in result:
                remaining = result.get("remaining", 0)
                print(f"\n{'='*60}")
                print(f"✅ {service.upper()} — Free (remaining today: {remaining})")
                print(f"{'='*60}")
                print(result["result"])
                return
            print(json.dumps(result, indent=2))
            return
    except urllib.error.HTTPError as e:
        if e.code == 429:
            body = json.loads(e.read())
            print(f"\n❌ Free limit reached (3/day)")
            print(f"\nTo continue, send {SERVICES[service]['cost']} USDC to:")
            print(f"  {WALLET}")
            print(f"  on {CHAIN} chain")
            print(f"\nThen retry with:")
            print(f"  curl -X POST {BASE_URL}/v1/{service} \\")
            print(f"    -H 'Content-Type: application/json' \\")
            print(f"    -H 'X-X402-Payment: YOUR_TX_HASH' \\")
            print(f"    -d '{json.dumps(data)}'")
            return
        elif e.code == 400:
            print(f"\n❌ Error: {e.read().decode()}")
            return
        else:
            print(f"\n❌ HTTP {e.code}: {e.read().decode()}")
            return

def main():
    if len(sys.argv) < 2 or sys.argv[1] in ("-h", "--help"):
        print("my-automaton CLI — x402 AI API Client")
        print(f"Wallet: {WALLET} ({CHAIN} · {TOKEN})")
        print()
        print("Services:")
        for name, info in SERVICES.items():
            print(f"  {name:12s} {info['cost']:4s}  {info['desc']}")
        print()
        print("Usage:")
        print(f"  python3 {sys.argv[0]} <service> \"<text>\"")
        print(f"  python3 {sys.argv[0]} <service> --file <path>")
        print(f"  python3 {sys.argv[0]} <service> --url <url>")
        print()
        print("Examples:")
        print(f"  python3 {sys.argv[0]} analyze \"AI is transforming how we code\"")
        print(f"  python3 {sys.argv[0]} review --file app.js")
        print(f"  python3 {sys.argv[0]} summarize --url https://example.com/doc")
        print()
        print("First 3 calls/day are FREE.")
        return
    
    service = sys.argv[1].lower()
    if service not in SERVICES:
        print(f"Unknown service: {service}")
        print(f"Available: {', '.join(SERVICES.keys())}")
        sys.exit(1)
    
    text = None
    code = None
    
    if "--file" in sys.argv:
        idx = sys.argv.index("--file") + 1
        if idx < len(sys.argv):
            path = sys.argv[idx]
            with open(path) as f:
                code = f.read()
            print(f"📄 Reading file: {path} ({len(code)} chars)")
    elif "--url" in sys.argv:
        idx = sys.argv.index("--url") + 1
        if idx < len(sys.argv):
            url = sys.argv[idx]
            print(f"🌐 Fetching URL: {url}")
            try:
                with urllib.request.urlopen(url, timeout=15) as resp:
                    content = resp.read().decode()
                    text = content[:5000]
                print(f"📄 Got {len(content)} chars, using first 5000")
            except Exception as e:
                print(f"❌ Could not fetch URL: {e}")
                sys.exit(1)
    else:
        text = " ".join(sys.argv[2:]) if len(sys.argv) > 2 else ""
        if not text:
            print("Please provide text or use --file/--url")
            sys.exit(1)
    
    call_api(service, text, code)

if __name__ == "__main__":
    main()
